package/gluon-hoodselector/luasrc/usr/lib/lua/hoodselector/util.lua

```diff
@@ -0,0 +1,191 @@
+-- return value: −1 if the ray from A to the right bisects the edge [BC] (the lower vortex of [BC]
+-- is not seen as part of [BC]);
+--                0 if A is on [BC];
+--               +1 else
+function M.crossProdTest(x_A,y_A,x_B,y_B,x_C,y_C)
+  if y_A == y_B and y_B == y_C then
+    if (x_B <= x_A and x_A <= x_C) or (x_C <= x_A and x_A <= x_B) then
+      return 0
+    end
+    return 1
+  end
+  if not ((y_A == y_B) and (x_A == x_B)) then
+    if y_B > y_C then
+      -- swap B and C
+      local h = x_B
+      x_B = x_C
+      x_C = h
+      h = y_B
+      y_B = y_C
+      y_C = h
+    end
+    if (y_A <= y_B) or (y_A > y_C) then
+      return 1
+    end
+    local Delta = (x_B-x_A) * (y_C-y_A) - (y_B-y_A) * (x_C-x_A)
+    if Delta > 0 then
+      return 1
+    elseif Delta < 0 then
+      return -1
+    end
+  end
+  return 0
+end
+
+-- Source with pseudocode: https://de.wikipedia.org/wiki/Punkt-in-Polygon-Test_nach_Jordan
+-- see also: https://en.wikipedia.org/wiki/Point_in_polygon
+-- let P be a 2D Polygon and Q a 2D Point
+-- return value:  +1 if Q within P;
+--               −1 if Q outside of P;
+--                0 if Q on an edge of P
+function M.pointInPolygon(poly, point)
+  local t = -1
+  for i=1,#poly-1 do
+    t = t * M.crossProdTest(point.lon,point.lat,poly[i][2],poly[i][1],poly[i+1][2],poly[i+1][1])
+    if t == 0 then break end
+  end
+  return t
+end
+
+-- Return hood from the hood file based on geo position or nil if no real hood could be determined
+-- First check if an area has > 2 points and is hence a polygon. Else assume it is a rectangular
+-- box defined by two points (south-west and north-east)
+function M.getHoodByGeo(jhood,geo)
+  for _, hood in pairs(jhood) do
+    if hood.domain_code ~= site.default_domain() then
+    for _, area in pairs(hood.domain.hoodselector.shapes) do
+      if #area > 2 then
+        if (M.pointInPolygon(area,geo) == 1) then
+          return hood
+        end
+      else
+        if ( geo.lat >= area[1][1] and geo.lat < area[2][1] and geo.lon >= area[1][2]
+          and geo.lon < area[2][2] ) then
+          return hood
+        end
+      end
+    end
+    end
+  end
+  return nil
+end
+
+function M.set_hoodconfig(geoHood)
+  if uci:get('gluon', 'core', 'domain') ~= geoHood.domain_code then
+    uci:set('gluon', 'core', 'domain', geoHood.domain_code)
+    uci:save('gluon')
+    uci:commit('gluon') -- necessary?
+    os.execute('gluon-reconfigure')
+    io.stdout:write("Set hood \""..geoHood.domain.domain_names[geoHood.domain_code].."\"\n")
+    return true
+  end
+  return false
+end
+
+-- Return the default hood in the hood list.
+-- This method can return the following data:
+-- * default hood
+-- * nil if no default hood has been defined
+function M.getDefaultHood(jhood)
+  for _, h in pairs(jhood) do
+    if h.domain_code == site.default_domain() then
+      return h
+    end
+  end
+  return nil
+end
+
+function M.restart_services()
```

Comment:
@NeoRaider : If that like you expect? I am not sure if I have all processes listed, which needs to be restarted.